R/Renderer.R

```diff
@@ -89,23 +111,23 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
   private = list(
     output_dir = character(0),
     # factory method
-    block2md = function(block) {
+    block2md = function(block, report_type) {
       if (inherits(block, "TextBlock")) {
-        private$textBlock2md(block)
+        private$textBlock2md(block, report_type)
       } else if (inherits(block, "RcodeBlock")) {
-        private$rcodeBlock2md(block)
+        private$rcodeBlock2md(block, report_type)
       } else if (inherits(block, "PictureBlock")) {
-        private$pictureBlock2md(block)
+        private$pictureBlock2md(block, report_type)
       } else if (inherits(block, "TableBlock")) {
-        private$tableBlock2md(block)
+        private$tableBlock2md(block, report_type)
       } else if (inherits(block, "NewpageBlock")) {
         block$get_content()
       } else {
         stop("Unknown block class")
       }
     },
```

Comment:
An argument is added to four methods but only one actually uses it.

Would it not be better to have `report_type` as a private field set by `renderRmd` and checked by `rcodeBlock2md`?

Reply:
That's a great idea! I can't believe I didn't think of it sooner. It would definitely streamline the code and improve efficiency. Thanks for suggesting this!